CRAYSAT-1662: Document automatic conversion of hyphens to underscores (#70)
Cray-HPE/docs-sat: docs/usage/sat_bootprep.md

on options that may be used with the `list-vars` subcommand, refer to the man page
for the `sat bootprep` subcommand.

#### Using Hyphens in HPC CSM Software Recipe Variables

> Variable names with hyphens are not allowed in Jinja2 expressions, because they
> will be parsed as an arithmetic expression rather than a single variable.
> In order to support product names with hyphens, `sat bootprep` will convert
> hyphens to underscores in all top-level keys in the software recipe variables,
> as well as any variables passed to the command using `--vars` or `--vars-file`.
> For example, to refer to the version of the product
> `slingshot-host-software` in the bootprep input file, you would write
> `"{{slingshot_host_software.version}}"`.

Reviewer: ```suggestion
Variable names with hyphens are not allowed in Jinja2 expressions because they
are parsed as an arithmetic expression instead of a single variable.
To support product names with hyphens, `sat bootprep` converts
hyphens to underscores in all top-level keys in the software recipe variables.
It also converts any variables passed to the command using `--vars` or `--vars-file`.
For example, to refer to the version of the product
`slingshot-host-software` in the bootprep input file, you would write
`"{{slingshot_host_software.version}}"`.
```

Author: Suggestion looks good to me, but worth working out the conversation in https://github.com/Cray-HPE/docs-sat/pull/70#discussion_r1092399837

> `"{{slingshot_host_software.version}}"`.

Reviewer: Is `"{{slingshot_host_software.version}}"` how `sat bootprep` converts `slingshot-host-software`? You've been talking about how the command will convert hyphens in the rest of the paragraph, but the example is suddenly talking about the user. I'm a little confused about the connection, so the example is unclear to me. Maybe I can help clarify this if you explain.

Author: Sure, in fairness it is pretty confusing. At a high level, SAT bootprep usually reads two files. The first file is a listing of product names mapped to their versions. I think this file is referred to as "software recipe variables" throughout this doc. As of Alice, this file can now contain product names with hyphens in them. Usually a user does not have to edit this file, in fact by default it is stored in a remote repository on the system rather than the local filesystem.

The second file is the "SAT bootprep input file" which contains more like the specification of images, configs, and session templates to create. The bootprep input file will refer to variables defined in the first, "product variables" file. When referring to product names that contain hyphens, you would write them with underscores.

For example, the recipe variables file could look like:
```
slingshot-host-software:
  version: 1.2.3
```

The bootprep input file would look something like:
```
images:
- name: my-slingshot-image-{{slingshot_host_software.version}}
...
```

Reviewer: @ekamin-hpe, how about wording like this?

```suggestion
When referring to a variable with hyphens in the bootprep input file, keep
this in mind. For example, to refer to the product version for 
`slingshot-host-software` in the bootprep input file, you would write
`"{{slingshot_host_software.version}}"`.
```

Author: That looks good to me. I committed the suggestion here.
